### PR TITLE
[Admin] Approve & Delete orders

### PR DESCRIPTION
**Pull request by ghost**
## Description
Approve and delete order
## Migrations
NO

## Impacted Areas in Application
List general components of the application that this PR will affect:

- routes/api.php

## SQL query and `explain`

`explain update `order_details` set `deleted_at` = "2018-05-23 07:32:34", `updated_at` = "2018-05-23 07:32:34" where `order_id` = 1 and `order_details`.`deleted_at` is null\G`

> *************************** 1. row ***************************
>            id: 1
>   select_type: UPDATE
>         table: order_details
>    partitions: NULL
>          type: range
> possible_keys: order_details_order_id_foreign
>           key: order_details_order_id_foreign
>       key_len: 4
>           ref: const
>          rows: 1
>      filtered: 100.00
>         Extra: Using where
> 1 row in set (0.00 sec)

` explain update `orders` set `deleted_at` = "2018-05-23 07:32:34", `updated_at` = "2018-05-23 07:32:34" where `id` = 1\G`

> *************************** 1. row ***************************
>            id: 1
>   select_type: UPDATE
>         table: orders
>    partitions: NULL
>          type: range
> possible_keys: PRIMARY
>           key: PRIMARY
>       key_len: 4
>           ref: const
>          rows: 1
>      filtered: 100.00
>         Extra: Using where
> 1 row in set (0.00 sec)

`explain update `orders` set `status` = false, `updated_at` = "2018-05-23 07:37:53" where `id` = 2\G`

> *************************** 1. row ***************************
>            id: 1
>   select_type: UPDATE
>         table: orders
>    partitions: NULL
>          type: range
> possible_keys: PRIMARY
>           key: PRIMARY
>       key_len: 4
>           ref: const
>          rows: 1
>      filtered: 100.00
>         Extra: Using where
> 1 row in set (0.00 sec)

### CODE REVIEW

File: app/Http/Controllers/Api/OrderController.php [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    *
+    * @return array status code
+    */
+    public function changeStatus($id)
+    {
+        $order = Order::find($id);
+        $order->status = !$order->status;
+        $order->save();
+        $data['status'] = (int) $order->status;
+        $data['msg'] = __('orders.admin.list.updated');
```
Comment (at-xinhnguyen) [inline app/Http/Controllers/Api/OrderController.php]: a đâu thấy em dùng 2 dòng này đâu
```
$data['status'] = (int) $order->status;
$data['msg'] = __('orders.admin.list.updated');
```

File: app/Http/Controllers/Admin/OrderController.php [hunk 2]
```diff
@@ -28,4 +30,27 @@ public function index(Request $request)
...
+     * @return \Illuminate\Http\Response
+     */
+    public function destroy($id)
+    {
+        try {
+            $order = Order::findOrFail($id);
+            $orderDetail = OrderDetail::query()->where('order_id', $id)->delete();
+            if ($orderDetail) {
+                $order->delete();
+            }
```
Comment (at-xinhnguyen) [inline app/Http/Controllers/Admin/OrderController.php]: em dùng `relationship` của `laravel` cho gọn , hoạt động như nhau lun.
            `$order = Order::findOrFail($id);
            $order->orderDetails()->delete();
            $order->delete();`

File: app/Http/Controllers/Admin/OrderController.php [hunk 3]
```diff
@@ -28,4 +30,27 @@ public function index(Request $request)
...
+     * @param int $id order id
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function destroy($id)
+    {
+        try {
+            $order = Order::findOrFail($id);
+            $orderDetail = OrderDetail::query()->where('order_id', $id)->delete();
+            if ($orderDetail) {
```
Comment (at-xinhnguyen) [inline app/Http/Controllers/Admin/OrderController.php]: chỗ này k cần check đâu. vì có hay k mình cũng `xoá`

**Review by ghost — CHANGES_REQUESTED** [review]
Please check my cmt

File: app/Http/Controllers/Admin/OrderController.php [hunk 4]
```diff
@@ -28,4 +30,27 @@ public function index(Request $request)
...
+                $order->delete();
+            }
+            session(['message' => __('orders.admin.list.deleted')]);
+        } catch (ModelNotFoundException $e) {
+            session(['message' => __('orders.admin.list.id_not_found')]);
```
[the remark this answers is not in the document]
Comment (at-phuongvu) [reply app/Http/Controllers/Admin/OrderController.php]: dạ chỗ này không cần ạ vì `finally` em đẩy về `orders.index` ở đó em `pull session message` ra nên dù thế nào nó cx lấy được ạ :D

File: app/Http/Controllers/Api/OrderController.php (hunk 1 above)
Comment (at-phuongvu) [reply app/Http/Controllers/Api/OrderController.php]: dạ 2 cái này em dùng bên js anh ơi

**Review by ghost — APPROVED** [review]
LGTM

**Pull request merged**